lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1672,9 +1674,72 @@ void RemoveExplicitModules(std::vector<std::string> &args) {
 
 } // namespace
 
-void SwiftASTContext::AddExtraClangArgs(const std::vector<std::string> &ExtraArgs) {
+/// LLDB wrapper for `clang::driver::applyOverrideOptions` (which implements
+/// CCC_OVERRIDE_OPTIONS behavior).
+static void applyOverrideOptions(std::vector<std::string> &args,
+                                 llvm::StringRef overrideOpts) {
+  if (overrideOpts.empty())
+    return;
+
+  // Convert input args to the type required by applyOverrideOptions.
+  llvm::SmallVector<const char *, 64> raw_args;
+  // Add placeholder clang executable, which applyOverrideOptions expects to be
+  // the first argument.
+  raw_args.push_back("clang");
+  for (const std::string &arg : args)
+    raw_args.push_back(arg.data());
+
+  // LLVM stream backed by a callback. This is used to redirect
+  // applyOverrideOptions logging to LLDB.
+  struct CallbackStream : public llvm::raw_ostream {
+    using callback_t = std::function<void(const char *, size_t)>;
+    callback_t m_callback;
+    uint64_t m_pos = 0;
+
+    CallbackStream(callback_t callback) : m_callback(callback) {}
+    ~CallbackStream() override { flush(); }
+
+    void write_impl(const char *Ptr, size_t Size) override {
+      m_callback(Ptr, Size);
+      m_pos += Size;
+    }
+
+    uint64_t current_pos() const override { return m_pos; }
+  };
+
+  // Perform the override operations.
+  llvm::StringSet<> savedStrings;
+  auto *log = GetLog(LLDBLog::Expressions);
```

Comment:
This should be the types log, since this is a Typesystem plugin (and also that's where everything else logs to in SwiftASTContext)